src/plugins/teleop/Teleop.cc

```diff
@@ -0,0 +1,274 @@
+Teleop::~Teleop()
+{
+}
+
+/////////////////////////////////////////////////
+void Teleop::LoadConfig(const tinyxml2::XMLElement *_pluginElement)
+{
+  if (this->title.empty())
+    this->title = "Teleop";
+
+  ignition::gui::App()->findChild
+    <ignition::gui::MainWindow *>()->QuickWindow()->installEventFilter(this);
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnTeleopTwist()
+{
+  ignition::msgs::Twist cmdVelMsg;
+
+  cmdVelMsg.mutable_linear()->set_x(
+      this->dataPtr->linearDir * this->dataPtr->linearVel);
+  cmdVelMsg.mutable_angular()->set_z(
+      this->dataPtr->angularDir * this->dataPtr->angularVel);
+
+  if (this->dataPtr->newTopic){
+    this->dataPtr->cmdVelPub = ignition::transport::Node::Publisher();
+    this->dataPtr->cmdVelPub =
+    this->dataPtr->node.Advertise<ignition::msgs::Twist>(this->dataPtr->topic);
+    this->dataPtr->newTopic = false;
+  }
+
+  this->dataPtr->cmdVelPub.Publish(cmdVelMsg);
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnTopicSelection(const QString& _topic)
+{
+  this->dataPtr->newTopic = true;
+  this->dataPtr->topic = _topic.toStdString();
+  ignmsg << "[OnTopicSelection]: topic: " << this->dataPtr->topic << std::endl;
+
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnLinearVelSelection(const QString& _velocity)
+{
+  this->dataPtr->linearVel = _velocity.toDouble();
+  ignmsg << "[OnlinearVelSelection]: linear velocity: "
+      << this->dataPtr->linearVel << std::endl;
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnAngularVelSelection(const QString& _velocity)
+{
+  this->dataPtr->angularVel = _velocity.toDouble();
+  ignmsg << "[OnlinearVelSelection]: angular velocity: "
+      << this->dataPtr->angularVel << std::endl;
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnKeySwitch(bool _checked)
+{
+  this->dataPtr->linearDir = 0;
+  this->dataPtr->angularDir = 0;
+  this->dataPtr->keyEnable = _checked;
+}
+
+/////////////////////////////////////////////////
+void Teleop::OnSlidersSwitch(bool _checked)
+{
+  if(_checked){
+    this->dataPtr->linearDir = 1;
+    this->dataPtr->angularDir = 1;
+    OnTeleopTwist();
+  }
+}
+
+/////////////////////////////////////////////////
+bool Teleop::eventFilter(QObject *_obj, QEvent *_event)
+{
+  if(_event->type() == QEvent::KeyPress && this->dataPtr->keyEnable == true)
+  {
+    QKeyEvent *keyEvent = static_cast<QKeyEvent*>(_event);
+    switch(keyEvent->key())
+    {
+      case Qt::Key_W:
+        this->dataPtr->linearState = keyLinear::forward;
+        break;
+      case Qt::Key_A:
+        this->dataPtr->angularState = keyAngular::left;
+        break;
+      case Qt::Key_D:
+        this->dataPtr->angularState = keyAngular::right;
+        break;
+      case Qt::Key_X:
+        this->dataPtr->linearState = keyLinear::backward;
+        break;
+      default:
+        ignmsg << "A non valid key was pressed" << std::endl;
```

Comment:
Delete this message to avoid extra verbosity

Reply:
Done